packages/kit/src/core/sync/write_types/index.js

```diff
@@ -251,18 +250,24 @@ function update_types(config, manifest_data, route) {
 /**
  * @param {import('types').PageNode} node
  * @param {string} outdir
- * @param {string} params
+ * @param {boolean} is_page
+ * @param {boolean} [all_pages_have_load]
  */
-function process_node(node, outdir, params) {
-	let data;
-	let load;
-	let server_load;
-	let errors;
+function process_node(node, outdir, is_page, all_pages_have_load = true) {
+	const params = `${is_page ? 'Route' : 'Layout'}Params`;
```

Comment:
this is a side-note, probably for a separate PR, but I've been wondering if we should just always spell out the params type, i.e.

```ts
PageLoad<{ id: string }>
```

rather than 

```ts
PageLoad<RouteParams>
```

It's obviously more repetitive, but it means that if you hover over the `params` object you see this...

<img width="393" alt="image" src="https://user-images.githubusercontent.com/1162160/186703223-e4c52a33-3d23-4be7-9edc-e43727459e76.png">

...instead of this, which is less useful:

<img width="473" alt="image" src="https://user-images.githubusercontent.com/1162160/186703464-51c4c3bc-2d92-41a5-b95a-deab35be74aa.png">

Reply:
that's a good idea yeah, let's do this in a separate PR.